plugins/mac-capture/mac-sck-audio-capture.m

```diff
@@ -0,0 +1,386 @@
+#include "mac-sck-common.h"
+
+const char *sck_audio_capture_getname(void *unused __attribute__((unused)))
+{
+	return obs_module_text("SCK.Audio.Name");
+}
+
+static void destroy_audio_screen_stream(struct screen_capture *sc)
+{
+	if (sc->disp) {
+		[sc->disp stopCaptureWithCompletionHandler:^(
+				  NSError *_Nullable error) {
+			if (error && error.code != 3808) {
+				MACCAP_ERR(
+					"destroy_audio_screen_stream: Failed to stop stream with error %s\n",
+					[[error localizedFailureReason]
+						cStringUsingEncoding:
+							NSUTF8StringEncoding]);
+			}
+			os_event_signal(sc->disp_finished);
+		}];
+		os_event_wait(sc->disp_finished);
+	}
+
+	if (sc->stream_properties) {
+		[sc->stream_properties release];
+		sc->stream_properties = NULL;
+	}
+
+	if (sc->disp) {
+		[sc->disp release];
+		sc->disp = NULL;
+	}
+
+	os_event_destroy(sc->disp_finished);
+	os_event_destroy(sc->stream_start_completed);
+}
+
+static void sck_audio_capture_destroy(void *data)
+{
+	struct screen_capture *sc = data;
+
+	if (!sc)
+		return;
+
+	obs_enter_graphics();
+
+	destroy_audio_screen_stream(sc);
+
+	obs_leave_graphics();
+
+	if (sc->shareable_content) {
+		os_sem_wait(sc->shareable_content_available);
+		[sc->shareable_content release];
+		os_sem_destroy(sc->shareable_content_available);
+		sc->shareable_content_available = NULL;
+	}
+
+	if (sc->capture_delegate) {
+		[sc->capture_delegate release];
+	}
+	[sc->application_id release];
+
+	pthread_mutex_destroy(&sc->mutex);
+	bfree(sc);
+}
+
+static bool init_audio_screen_stream(struct screen_capture *sc)
+{
+	SCContentFilter *content_filter;
+
+	sc->stream_properties = [[SCStreamConfiguration alloc] init];
+	os_sem_wait(sc->shareable_content_available);
+
+	SCDisplay * (^get_target_display)() = ^SCDisplay *()
+	{
+		__block SCDisplay *target_display = nil;
+		[sc->shareable_content.displays
+			indexOfObjectPassingTest:^BOOL(
+				SCDisplay *_Nonnull display, NSUInteger idx,
+				BOOL *_Nonnull stop) {
+				if (display.displayID == sc->display) {
+					target_display = sc->shareable_content
+								 .displays[idx];
+					*stop = TRUE;
+				}
+				return *stop;
+			}];
+		return target_display;
+	};
+
+	void (^set_display_mode)(struct screen_capture *, SCDisplay *) = ^void(
+		struct screen_capture *sc2, SCDisplay *target_display) {
+		CGDisplayModeRef display_mode =
+			CGDisplayCopyDisplayMode(target_display.displayID);
+		[sc2->stream_properties
+			setWidth:CGDisplayModeGetPixelWidth(display_mode)];
+		[sc2->stream_properties
+			setHeight:CGDisplayModeGetPixelHeight(display_mode)];
+		CGDisplayModeRelease(display_mode);
+	};
+
+	switch (sc->capture_type) {
+	case ScreenCaptureAudioDesktopStream: {
+		SCDisplay *target_display = get_target_display();
+
+		content_filter = [[SCContentFilter alloc]
+			 initWithDisplay:target_display
+			excludingWindows:[[NSArray alloc] init]];
+
+		set_display_mode(sc, target_display);
```

Comment:
Is this even necessary when we're just interested in audio samples?

Reply:
Yeah ``set_display_mode`` definitely seems unnecessary, will remove.